crates/rattler_lock/src/options.rs

```diff
@@ -0,0 +1,20 @@
+use rattler_solve::{ChannelPriority, SolveStrategy};
+
+/// Options that were used during the resolution of the packages stored in the
+/// lock-file. These options strongly influence the outcome of the solve and are
+/// therefore stored along with the locked packages.
+#[derive(Debug, Clone, PartialEq, Eq, Default, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct ResolverOptions {
+    /// The strategy that was used to solve the dependencies.
+    #[serde(default, skip_serializing_if = "crate::utils::serde::is_default")]
+    pub strategy: SolveStrategy,
+
+    /// The channel priority that was used to solve the dependencies.
+    #[serde(default, skip_serializing_if = "crate::utils::serde::is_default")]
+    pub channel_priority: ChannelPriority,
```

Comment:
What is the advantage of this approach over wrapping them into an `Option` like with `exclude_newer`?

Reply:
I think we should have some default values here and not have the user have to figure out what those are.

Exclude-newer is wrapped in an option because otherwise we cannot represent "not" exclude-newer.

Reply:
Okay!